packages/playwright-core/src/server/injected/ariaSnapshot.ts

```diff
@@ -275,22 +275,36 @@ function matchesNode(node: AriaNode | string, template: AriaTemplateNode, depth:
       return false;
     if (!matchesText(node.props.url, template.props?.url))
       return false;
-    if (!containsList(node.children || [], template.children || [], depth))
+    if ((isDeepStrict || template.containerMode === 'deep-equal') && !listEqual(node.children || [], template.children || [], true))
```

Comment:
Can I opt-out of `deep-equal` by setting `contain` somewhere in the subtree? If so, this one should check that `template.containerMode !== 'contain'`. If not, we should probably throw in such a case.

Reply:
Sure, I can account for this.